pkg/controller/tsp/metric.go

```diff
@@ -0,0 +1,132 @@
+package tsp
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/gocrane/api/prediction/v1alpha1"
```

Comment:
It is better to have a custom package name for this. 
If you want to upgrade the interface, you  just need to change this import package and no need to change the codes.